src/bun.js/api/server.zig

```diff
@@ -3213,11 +3213,17 @@ fn NewRequestContext(comptime ssl_enabled: bool, comptime debug_mode: bool, comp
                 status;
 
             var needs_content_type = true;
+            var content_type_needs_free = false;
+
             const content_type: MimeType = brk: {
                 if (response.init.headers) |headers_| {
                     if (headers_.fastGet(.ContentType)) |content| {
                         needs_content_type = false;
-                        break :brk MimeType.byName(content.slice());
+
+                        var content_slice = content.toSlice(this.allocator);
+                        defer content_slice.deinit();
+
+                        break :brk MimeType.init(content_slice.slice(), content_slice.allocator.get(), &content_type_needs_free);
```

Comment:
This is not the correct allocator:

```suggestion
                        break :brk MimeType.init(content_slice.slice(), this.allocator, &content_type_needs_free);
```

Reply:
If I pass `this.allocator` will not return the same allocator I pass on?
https://github.com/oven-sh/bun/blob/main/src/bun.js/bindings/bindings.zig#L766-L791
Looks like it would do that
https://github.com/oven-sh/bun/blob/main/src/NullableAllocator.zig
The ideia was only allocate if necessary, if i pass `this.allocator` will always allocate.
I can also do like `if (content_slice.isNull()) null else this.allocator` but dont look necessary.

Reply:
that makes sense, but we can't use the StringImpl allocator to allocate any memory. It only increments the stringImpl's reference count.

Reply:
fixed in https://github.com/oven-sh/bun/pull/11695/commits/c6b26e1bdedf5242fa14bb4a5bbabc6163cd6aec